CMakeLists.txt

```diff
@@ -2196,6 +2196,9 @@ include(HPX_SetupAsio)
 include(HPX_SetupAllocator)
 include(HPX_SetupHwloc)
 
+# Setup NVIDIA's stdexec
+include(HPX_SetupStdexec)
```

Comment:
This probably should be included only if compiling with C++20 or higher. We should also introduce an `HPX_WITH_...` option to explicitly enable pulling it in.

Reply:
Yess totally, and more nuances are to come 
(this is a bare minimum pr for now)

Reply:
Do you think it would make sense to perform this check when parsing the arguments in the [main CMakeLists.txt](https://github.com/STEllAR-GROUP/hpx/pull/6431/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR611)?

I did not spot any convention for that sort of thing in the CMake files.

Reply:
> Do you think it would make sense to perform this check when parsing the arguments in the [main CMakeLists.txt](https://github.com/STEllAR-GROUP/hpx/pull/6431/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR611)?
> 
> I did not spot any convention for that sort of thing in the CMake files.

Yah, we don't have conventions. Let's move as much as possible out of the main CMakeLists.txt, it's overcrowded already.